tests/test_pipeline_conversion.py

```diff
@@ -50,3 +50,26 @@ def test_pipeline_conversion_override_inputs(pipeline_overridden_config: Config)
     assert overridden["template"]["inputs"].get('training-images', [])[0] == 'overridden node image'
     assert len(overridden["template"]["inputs"].get('training-images', [])) == 1
     assert len(overridden["template"]["parameters"].items()) == 3
+
+
+def test_pipeline_parameter_conversion(pipeline_with_parameters_config):
+    parameter_name = "id"
+    for _name, pipe in pipeline_with_parameters_config.pipelines.items():
+        result = PipelineConverter(
+            config=pipeline_with_parameters_config,
+            commit_identifier="latest",
+        ).convert_pipeline(pipe)
+        assert isinstance(result["parameters"], dict)
+        assert result["parameters"][parameter_name]
+
+        parameter = result["parameters"][parameter_name]
+        assert parameter["config"]["targets"]
+        assert "target" not in parameter["config"]
+        assert isinstance(parameter["config"]["targets"], list)
+
+        # parameter expression check
+        if _name == "Example Pipeline with Parameters":
+            assert parameter["expression"] == 123
+        else:
+            # When a parameter's default is missing, it should be an empty string
+            assert parameter["expression"] == ""
```

Comment:
This works as well, but it speaks for a different cause-effect relation. The name of pipeline doesn't effect the default value straight. You should dig the actual parameter config from pipeline's config and use that to determinate what value the expression should have:

```suggestion
        parameter_config = next(param for param in pipe.parameters if param["name"] == parameter_name)
        if parameter_config is None:
            assert parameter["expression"] == ""
        else:
            assert parameter["expression"] == parameter_config.default
```

Reply:
Got the point. Changed it.